sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -575,6 +575,8 @@ def __setitem__(self, key, value):
     else:
       raise NotImplementedError(key)
 
+  align = frame_base._elementwise_method('align')
```

Comment:
Is this safe to implement with `elementwise_method`? (need to justify this to myself)

I think it will work for most cases.. the resulting expression will require `partitionings.Index` since we do that if any arg is a DeferredBase, and `other` should always be one. That should handle axis='index'. Also all of the join modes should work as intended within Index partitions.

I think actually for `axis='columns'` Index partitioning is _too_ restrictive. We could do that without any partitioning. Is that right?

Looking at the remaining args:
- `copy`: I think copy=False won't work since we can't predict if the operation is inplace or not, it depends on the data.
- `fill_value`: Works trivially
- `method`: non-default options are order-sensitive
- `limit`: I don't think we can support this correctly right now without Singleton partitioning
- `level`, `fill_axis`, `broadcast_axis`: I'm not actually sure what these are doing.

Can we reject the options that won't work?

Reply:
Good call. Done.